lib/crowdin/adapter.rb

```diff
@@ -84,27 +84,34 @@ def translations(language)
         approved_file_ids = file_status_list.select { |f| f['approvalProgress'] == 100 }.map { |f| f['file_id'] }
 
         # Fetch translations for each approved file
-        to_return = translations_for_files(approved_file_ids, language)
-        if to_return.failure.present? && to_return.failure.is_a?(CrowdIn::FileMethods::FilesError)
-          @files_to_cleanup = approved_file_ids - to_return.failure.errors_by_file.keys
```

Comment:
removing the use of this @files_to_cleanup instance variable. Now the files to cleanup are computed from the argument to the cleanup_translations method, which is overall much cleaner.